syntax = "proto3";

package temperature;

// Request and response messages for Celsius to Fahrenheit conversion
message CelsiusToFahrenheitRequest {
  double temperature_celsius = 1;
}

message CelsiusToFahrenheitResponse {
  double temperature_fahrenheit = 1;
}

// Request and response messages for Fahrenheit to Celsius conversion
message FahrenheitToCelsiusRequest {
  double temperature_fahrenheit = 1;
}

message FahrenheitToCelsiusResponse {
  double temperature_celsius = 1;
}

// Service definition for temperature conversion
service TemperatureConversion {
  rpc ConvertCelsiusToFahrenheit (CelsiusToFahrenheitRequest) returns (CelsiusToFahrenheitResponse) {}
  rpc ConvertFahrenheitToCelsius (FahrenheitToCelsiusRequest) returns (FahrenheitToCelsiusResponse) {}
}
